src/core/meshcop/dtls.cpp

```diff
@@ -56,6 +56,11 @@
 namespace ot {
 namespace MeshCoP {
 
+const mbedtls_ecp_group_id Dtls::sCurves[] = {MBEDTLS_ECP_DP_SECP256R1, MBEDTLS_ECP_DP_NONE};
+#if defined(MBEDTLS_KEY_EXCHANGE__WITH_CERT__ENABLED)
```

Comment:
nit: the rest of this file uses `#ifdef`
```suggestion
#ifdef MBEDTLS_KEY_EXCHANGE__WITH_CERT__ENABLED
```